### DISCUSSION TITLE
Cannot compile Caribou/SofaCaribou

### DISCUSSION THREAD

**Discussion by ateixeira0163** [jnbrunet/caribou#54] [Q&A] [answered]
I spent already way too much time trying to make it work, so I'm calling for help :) 
I just did all mentioned in : https://caribou.readthedocs.io/en/latest/installation.html#compiling but when trying to do 'cmake ..' I always seems to get the same error : 

<pre>CMake Error at /home/ateixeira/Sofa/src/build/cmake/GTestConfig.cmake:35 (include):
  include could not find requested file:

    /home/ateixeira/Sofa/src/build/cmake/GTestTargets.cmake
Call Stack (most recent call first):
  /home/ateixeira/Sofa/src/build/cmake/SofaFrameworkConfig.cmake:91 (find_package)
  cmake/FindSOFA.cmake:19 (find_package)
  CMakeLists.txt:56 (find_package)

CMake Error at /home/ateixeira/Sofa/src/build/cmake/SofaFrameworkConfig.cmake:99 (include):
  include could not find requested file:

    /home/ateixeira/Sofa/src/build/cmake/SofaFrameworkTargets.cmake
Call Stack (most recent call first):
  cmake/FindSOFA.cmake:19 (find_package)
  CMakeLists.txt:56 (find_package)

CMake Error at /home/ateixeira/Sofa/src/build/cmake/SofaFrameworkConfig.cmake:104 (include):
  include could not find requested file:

    /home/ateixeira/Sofa/src/build/cmake/SofaFrameworkTargets.cmake
Call Stack (most recent call first):
  cmake/FindSOFA.cmake:19 (find_package)
  CMakeLists.txt:56 (find_package)
</pre>

the command used was : 
`cmake -DCMAKE_PREFIX_PATH="$SOFA_INSTALL/lib/cmake;$SP3_INSTALL/lib/cmake" ..`

with `$SP3_INSTALL` and `$SOFA_INSTALL` well defined (the path to PluginTargets.cmake and SofaFrameworkTargets.cmake exists). The only weird thing is that these files are not in the same place as said. More exactly, the path to these files are : `$~/Sofa/src/build/SofaKernel/SofaFramework/CMakeFiles/Export/lib/cmake/SofaFramework/SofaFrameworkTargets.cmake` and `$~/Sofa/src/build/applications/plugins/SofaPython3/Plugin/CMakeFiles/Export/lib/cmake/SofaPython3/PluginTargets.cmake`

Thanks in advance :)

**Comment by jnbrunet** [comment, marked as answer]
Hi @ateixeira0163 ,

Make sure you are using the **installation path** of both SOFA and SofaPython3. The **installation path** is the path to the directory created during the `make install` command. It is usually the build directory path, followed by the "install" directory. In your case, it looks like SofaPython3 is compiled alongside SOFA, so it is even simpler. Something like this should work:

```shell
~ $ cd ~/Sofa/src/build
~/Sofa/src/build $  cmake -DPLUGIN_SOFAPYTHON3=ON ..
~/Sofa/src/build $  make
~/Sofa/src/build $  make install
~/Sofa/src/build $  export SOFA_ROOT=~/Sofa/src/build/install   # See the "/install" here?

~ $ cd ~/caribou/src/build      # Or whatever your caribou build directory path is...
~/caribou/src/build $ cmake ..  # It should find SOFA and SofaPython3 correctly
~/caribou/src/build $ make
```

If you noticed, I used the environment variable `SOFA_ROOT` instead of `SOFA_INSTALL`.  This will be updated soon in the doc, but using this will allow Caribou to find SOFA and its plugins (like SofaPython3) automatically without having to manually change the `CMAKE_PREFIX_PATH` variable.
